modules/codegen/src/smithy4s/codegen/internals/IR.scala

```diff
@@ -250,8 +250,10 @@ private[internals] object Hint {
   case class Constraint(tr: Type.Ref, native: Native) extends Hint
   case class Protocol(traits: List[Type.Ref]) extends Hint
   case class Default(typedNode: Fix[TypedNode]) extends Hint
-  case class Documentation(docString: String, memberDocs: Map[String, String])
-      extends Hint
+  case class Documentation(
```

Comment:
I think it may be easier to pre-split : 

``` 
case class Documentation(docLines: List[String], memberDocLines: Map[String, List[String]]) extends Hint 
```

Reply:
With the optionality stuff (docLines or memberDocLines might be missing/empty), do I wrap both properties in an Option?

`docsLines: Option[List[String]], memberDocLines: Option[Map[String, List[String]]`

Reply:
probably not, `Nil`/`Map.empty` would be the equivalent. I'm thinking to either do options and change the types inside to be `NonEmpty(...)`, or keep it like in @Baccata's suggestion.